src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Resolve.java

```diff
@@ -3686,7 +3686,8 @@ class ConstructorReferenceLookupHelper extends ReferenceLookupHelper {
                 List<Type> typeargtypes, MethodResolutionPhase maxPhase) {
             super(referenceTree, names.init, site, argtypes, typeargtypes, maxPhase);
             if (site.isRaw()) {
-                this.site = new ClassType(site.getEnclosingType(), site.tsym.type.getTypeArguments(), site.tsym, site.getMetadata());
+                this.site = new ClassType(site.getEnclosingType(),
+                        !site.tsym.isInner() ? site.tsym.type.getTypeArguments() : List.nil(), site.tsym, site.getMetadata());
                 needsInference = true;
```

Comment:
so here it seems like we are still doing dimond inference but check that we are creating a new class type with no arguments so no inference will be done in this case, but we are still setting `needsInference` to true in order to reuse the code in Resolve::findDiamond that creates a new MethodSymbol for the constructor with the right result type instead of `void`